metron-platform/metron-elasticsearch/src/test/java/org/apache/metron/elasticsearch/integration/ElasticsearchSearchIntegrationTest.java

```diff
@@ -97,48 +118,81 @@ protected static InMemoryComponent startIndex() throws Exception {
-        .setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
-    JSONArray broArray = (JSONArray) new JSONParser().parse(broData);
-    for (Object o : broArray) {
-      JSONObject jsonObject = (JSONObject) o;
-      IndexRequestBuilder indexRequestBuilder = es.getClient()
-          .prepareIndex("bro_index_2017.01.01.01", "bro_doc");
-      indexRequestBuilder = indexRequestBuilder.setId((String) jsonObject.get("guid"));
-      indexRequestBuilder = indexRequestBuilder.setSource(jsonObject.toJSONString());
-      indexRequestBuilder = indexRequestBuilder
-          .setTimestamp(jsonObject.get("timestamp").toString());
-      bulkRequest.add(indexRequestBuilder);
+    es.getClient().admin().indices().prepareCreate(snortIndex)
+            .addMapping("snort_doc", JSONUtils.INSTANCE.toJSON(snortTemplate.get("mappings"), false)).get();
+
+    // setup the classes required to write the test data
+    AccessConfig accessConfig = createAccessConfig();
+    ElasticsearchClient client = ElasticsearchUtils.getClient(createGlobalConfig());
+    ElasticsearchRetrieveLatestDao retrieveLatestDao = new ElasticsearchRetrieveLatestDao(client);
+    ElasticsearchColumnMetadataDao columnMetadataDao = new ElasticsearchColumnMetadataDao(client);
+    ElasticsearchRequestSubmitter requestSubmitter = new ElasticsearchRequestSubmitter(client);
+    ElasticsearchUpdateDao updateDao = new ElasticsearchUpdateDao(client, accessConfig, retrieveLatestDao);
+    ElasticsearchSearchDao searchDao = new ElasticsearchSearchDao(client, accessConfig, columnMetadataDao, requestSubmitter);
+
+    // write the test documents for Bro
+    List<String> broDocuments = new ArrayList<>();
+    for (Object broObject: (JSONArray) new JSONParser().parse(broData)) {
+      broDocuments.add(((JSONObject) broObject).toJSONString());
     }
-    JSONArray snortArray = (JSONArray) new JSONParser().parse(snortData);
-    for (Object o : snortArray) {
-      JSONObject jsonObject = (JSONObject) o;
-      IndexRequestBuilder indexRequestBuilder = es.getClient()
-          .prepareIndex("snort_index_2017.01.01.02", "snort_doc");
-      indexRequestBuilder = indexRequestBuilder.setId((String) jsonObject.get("guid"));
-      indexRequestBuilder = indexRequestBuilder.setSource(jsonObject.toJSONString());
-      indexRequestBuilder = indexRequestBuilder
-          .setTimestamp(jsonObject.get("timestamp").toString());
-      bulkRequest.add(indexRequestBuilder);
+    es.add(updateDao, broIndex, "bro", broDocuments);
+
+    // write the test documents for Snort
+    List<String> snortDocuments = new ArrayList<>();
+    for (Object snortObject: (JSONArray) new JSONParser().parse(snortData)) {
+      snortDocuments.add(((JSONObject) snortObject).toJSONString());
     }
-    BulkResponse bulkResponse = bulkRequest.execute().actionGet();
-    if (bulkResponse.hasFailures()) {
-      throw new RuntimeException("Failed to index test data");
+    es.add(updateDao, snortIndex, "snort", snortDocuments);
```

Comment:
We are now using `ElasticsearchComponent.add` to load the test data here.  This change matches what already exists in the other ES integration tests.